pysynchronizer/errors.py

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (c) 2013 Citrix Systems, Inc.
+# Copyright (c) 2021 Daniel P. Smith, Apertus Solutions LLC
+# 
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+# 
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+
+class Error(Exception):
+    """Base class for other exceptions"""
+
+class ConfigError(Error):
```

Comment:
This is unused.  I didn't check for the other ones.

Reply:
actually, several are probably unused since I don't think they were all used in the original sync-client. I just pulled all the errors and I don't think it is unreasonable not to keep them. Should this code continue to expand, there will be a suite of errors available so that an appropriate one may be raised and caught.

Reply:
ConfigError is now used?  But it's for `raise ConfigError("VMDisk(%s) has an invalid path (%s)" % (self.id, path))` which doesn't match """sync-client configuration is not valid"""

My preference would be to only add used code.  @crogers1 gave a :+1: so I guess it's okay to stay.

Reply:
Turns out that was a bug in my code which I fixed it to use the error I originally intended. And yes the comment doesn't align exactly but I was just using the already existing errors. A new one named something like SyncRuntimeError could be created if that is the desire.